### ISSUE TITLE
New Event: The Well

### ISSUE THREAD

**Issue by wang429** [dbjorge/jorbs-spire-mod#353]
Original idea by Fr33ly on discord. Expanded by MetaBoy and CrumpocolypseSurviver

TL;DR:
Costs: HP, Max HP, Potion 
Rewards: eye of the storm, relic that allows seeing the next random memory

Chat dump below for some of the initial discussion.
> Fr33lyToday at 12:03 PM
I had an idea for an event
but it might be hard to code
you encounter a magical well and can choose to sacrifice a potion (dump it in) and lose some hp to recieve an additional eye of the storm card
if you choose to do so, you then receive another prompt to use the empty potion to bottle some of the well water (and lose some max HP) or just leave
if you bottle the well water you recieve a relic (something like stormwater phial) which makes it so you can see which memory you will random into if you cast eye of the storm
brosmikeToday at 1:54 PM
That sounds like a neat idea! I don't think it should be prohibitively difficult to code. Would probably not get to it until after the workshop release, though
Fr33lyToday at 1:55 PM
The part which concerns me is how do you let the player decide which potion to throw away? I thought that maybe you can just discard a potion while in the event menu (which has the added upside of being able to acquire more eye of the storms if you so choose), but that seems even harder to code.
brosmikeToday at 1:56 PM
I would just do different event menu buttons labeled with the potions you have, like how that event where you fall and have to choose a card to lose works
Fr33lyToday at 1:58 PM
awesome! Hope jorbs likes it enough to make it into the mod! I imagine it'd be too strong for a act1 event, maybe act2?
brosmikeToday at 1:59 PM
My guesses at balance decisions rarely match Jorbs', I'll leave that for him
CrumpocolypseSurviverToday at 3:37 PM
Instead of seeing which you would random into, why not make it a choice between two memories chosen at random?
MetaBoyToday at 3:46 PM
@Fr33ly I really like this but I think it would work better if this was a choice between either +1 storm or the relic instead of being able to get both.
This is how most events like that work too
Fr33lyToday at 3:49 PM
Yeah, I like that way more too actually!
MetaBoyToday at 3:49 PM
@CrumpocolypseSurviver I like the idea of it just giving you vision without any other benefit
Are there any other random memory cards besides eots?
Fr33lyToday at 3:52 PM
Wandering mind
MetaBoyToday at 3:55 PM
It could be a general “see next random memory” effect.
The question is, how does the game decide, for example, when this next memory value changes and where is it stored
It becomes its own mechanic
Fr33lyToday at 3:57 PM
After every card played? Could make it synergistic with coil decks/material component decks
although it could be a hassle checking your card after every card you play, so maybe after every time you acquire a new memory?
MetaBoyToday at 3:57 PM
You could do it like this: there is a single “next memory”. It only changes when you random/directly remember that memory. Otherwise it stays the same
Anything that randoms into a memory uses that value
If it changed every card played you could abuse it very easily, right?
Just play cards until it’s the one you want
Fr33lyToday at 4:00 PM
I think end of turn and acquire memory would be good events that would trigger a new random outcome
end of turn affects things like runic pyramid
MetaBoyToday at 4:01 PM
Yeah that works
Fr33lyToday at 4:01 PM
acquire new memory makes it so you want more Remember X cards to interact with the relic
MetaBoyToday at 4:02 PM
I like each turn+every new memory combined
Gives you two angles to control it
Next memory would be shown in relic ui
And flashed when you hold a card that randoms into a memory
BidoofBoi (he/him)Today at 4:05 PM
hmm, would it be better if it just lightly highlighted the next memory on the wheel?
MetaBoyToday at 4:05 PM
Yeah, highlighted or flashed or something
Fr33lyToday at 4:06 PM
i think messing with the memory wheel might confuse people (with the current remembered memory particles there too)
I like the idea of the relic turning into the memory icon that you will random into while you are hovering the card
but that might be hard to code
at this point, whichever is easiest to implement gets my vote :smile:
MetaBoyToday at 4:06 PM
Do “remember” cards show you the memory on the wheel when you hold them?
Fr33lyToday at 4:07 PM
If not they should!
MetaBoyToday at 4:07 PM
If yes this should do the same for random memory cards
Fr33lyToday at 4:07 PM
agree, that would be a great way to implement this
BobertgToday at 4:08 PM
there's also the potion Liquid Virtue that grants 2 random virtue clarities
MetaBoyToday at 4:10 PM
I think the idea is it only gives you a glimpse to the next one, so the interaction here would be you knowing one of the memories you’ll get
Ah, it’s only virtues
Then it only works if the next one is a virtue I guess
BobertgToday at 4:12 PM
what about rot which removes a random clarity?
MetaBoyToday at 4:14 PM
Yeah it gets very messy if there is only one value. I guess you would track a separate one for each type of random memory?
Should probably just ignore rot though
It’s different enough
hmm
BobertgToday at 4:19 PM
could just ignore clarities altogether
MetaBoyToday at 4:23 PM
Yeah
BobertgToday at 4:34 PM
would this event be wanderer-only? would it perhaps make sense for the event to still occur but actions change based on character?
MetaBoyToday at 4:35 PM
Dried up well for cull
Just a normal (not magical) well for explorer
CrumpocolypseSurviverToday at 4:36 PM
explorer could get some kind of heal, like a safe rest spot
MetaBoyToday at 4:36 PM
It’s wanderer who polluted the water with his magic over time
